python/ray/dag/collective_node.py

```diff
@@ -37,22 +37,21 @@ def __init__(
         op: _CollectiveOp,
         transport: Optional[Union[str, GPUCommunicator]] = None,
     ):
-        self._input_nodes: List[DAGNode] = input_nodes
```

Comment:
If we persist input_nodes in `_CollectiveOperation`, all input_nodes will be added to the [upstream_nodes](https://github.com/ray-project/ray/blob/f44890f90284c41c4d26a2d1e3b36b3ce4a8f4ff/python/ray/dag/dag_node.py#L88) when building the DAG. However, not all `input_nodes` belong to the args or kwargs of the DAG node. This could potentially cause issues when compiling the graph. cc @dengwxn